pandas/tests/io/parser/test_common.py

```diff
@@ -1916,6 +1916,24 @@ def test_filename_with_special_chars(all_parsers):
         tm.assert_frame_equal(result, df)
 
 
+def test_read_csv_memory_growth_chunksize(all_parsers):
+    # see gh-24805
+    #
+    # Let's just make sure that we don't crash
+    # as we iteratively process all chunks.
```

Comment:
maybe use psutil to track memory?

Reply:
Cool.  Let me add that test quickly then.

Reply:
its a test that doesn't assert anything, and where the failure mode is crashing.  how much RAM does a machine need before the test will incorrectly pass?

Reply:
@jbrockmendel : I have 16 GB of RAM on my machine, and this test failed before the patch.

Reply:
on a 128gb machine I'm getting failures in 2/3 cases pre-patch.  is that expected, or should it be 3/3?

It's not that I doubt the correctness of the patch, just that I'm really uncomfortable with the failure mode of the test.

Reply:
@jbrockmendel : 2/3 failures pre-patch is what you should get.

> just that I'm really uncomfortable with the failure mode of the test.

I understand your concern, though such a failure mode is not uncommon for our CSV tests, especially for it's impacting low-level C engine code.

If you feel like you want to add something to do this, by all means.